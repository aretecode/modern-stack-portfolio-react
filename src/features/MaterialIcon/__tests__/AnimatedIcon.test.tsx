import * as React from 'react'
import { render, fireEvent } from 'react-testing-library'
import {
  UP_ARROW_PATH_VALUE,
  DOWN_ARROW_PATH_VALUE,
  AnimatedArrowIcon,
} from '../AnimatedArrowIcon'

describe('AnimatedArrowIcon', () => {
  it('exports the same type of values', () => {
    expect(typeof UP_ARROW_PATH_VALUE).toEqual('string')
    expect(typeof DOWN_ARROW_PATH_VALUE).toEqual('string')
    expect(typeof AnimatedArrowIcon).toEqual('function')
  })

  describe('rendering', () => {
    /**
     * could also test initial path values
     */
    it('should have the values in the correct direction', async () => {
      const view = <AnimatedArrowIcon />
      const { container, rerender } = render(view)
      // react-testing-library wraps with a div
      // svg > title,polygon,path > animation
      const svgElement = container.firstChild!
      const polygonElement = svgElement.children[1]!
      const animationElement = polygonElement.firstChild

      // could also do: polygonElement.getAttribute('points')
      expect(polygonElement).toMatchSnapshot('initial')

      fireEvent.click(svgElement)
      rerender(view)

      const afterClickingValues = animationElement.getAttribute('values')
      // @todo - rendered value is changed slightly
      // expect(afterClickingValues).toEqual(DOWN_ARROW_PATH_VALUE)
      expect(afterClickingValues).toMatchSnapshot('down')

      // click again to render v^
      fireEvent.click(svgElement)
      rerender(view)

      const afterSecondClickValues = animationElement.getAttribute('values')
      expect(afterSecondClickValues).toMatchSnapshot('up')
    })
  })
})
