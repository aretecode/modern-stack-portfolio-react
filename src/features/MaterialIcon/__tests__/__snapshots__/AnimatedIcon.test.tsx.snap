// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnimatedArrowIcon rendering should have the values in the correct direction: down 1`] = `"12.5 13.37 18.4 8 20 10.1 12.5 17 5 10.1 6.63 8.04;12.5472 8 19.6230229 12.8615485 18.1969774 14.8285039 12.3972809 11.1356625 6.6896367 14.8285039 5 12.8615485;12.4846802 8 19.9693604 14.8939819 18.386673 16.9936981 12.4846802 11.6221466 6.61653912 16.9414647 5 14.8939819"`;

exports[`AnimatedArrowIcon rendering should have the values in the correct direction: initial 1`] = `
<polygon
  points="12.5 13.37 18.4 8 20 10.1 12.5 17 5 10.1 6.63 8.04"
>
  <animate
    attributeName="points"
    begin="0s"
    dur="0ms"
    fill="freeze"
    values=""
  />
</polygon>
`;

exports[`AnimatedArrowIcon rendering should have the values in the correct direction: up 1`] = `"12.5 13.37 18.4 8 20 10.1 12.5 17 5 10.1 6.63 8.04;12.5472 8 19.6230229 12.8615485 18.1969774 14.8285039 12.3972809 11.1356625 6.6896367 14.8285039 5 12.8615485;12.4846802 8 19.9693604 14.8939819 18.386673 16.9936981 12.4846802 11.6221466 6.61653912 16.9414647 5 14.8939819"`;
